static/src/js/pages/KeywordManagerPage/__tests__/KeywordManagerPage.test.jsx

```diff
@@ -328,6 +332,232 @@ describe('KeywordManagerPage component', () => {
+      })
+
+      // Ensure the main publish button is still disabled after closing the success modal
+      expect(publishButton).toBeDisabled()
+    })
+
+    test('should show success modal after successful publish and close it using both toggleModal and OK button', async () => {
+      publishKmsConceptVersionMock.mockResolvedValue()
+      const { user } = setup()
+
+      // Open the publish modal
+      const publishButton = screen.getByRole('button', { name: /publish new keyword version/i })
+      await user.click(publishButton)
+
+      // Wait for the modal to appear
+      const modal = await screen.findByTestId('custom-modal')
+      expect(modal).toBeInTheDocument()
+
+      // Find the version name input within the modal
+      const versionNameInput = within(modal).getByRole('textbox')
+      expect(versionNameInput).toBeInTheDocument()
+
+      // Enter a new version name
+      await user.type(versionNameInput, 'NewVersion')
+
+      // Click the publish button in the modal
+      const modalPublishButton = within(modal).getByRole('button', { name: /publish/i })
+      await user.click(modalPublishButton)
+
+      // Check that the success modal is shown
+      let successModal
+      await waitFor(() => {
+        successModal = screen.getByTestId('custom-modal')
+        expect(successModal).toBeInTheDocument()
+      })
+
+      expect(successModal).toHaveTextContent(/initiated new published version/i)
+      expect(successModal).toHaveTextContent(/refresh browser after a few minutes/i)
+
+      // Close the modal using the OK button (onClick handler)
+      const okButton = screen.getByRole('button', { name: /ok/i })
+      await user.click(okButton)
+
+      // Check that the success modal is closed
+      await waitFor(() => {
+        expect(screen.queryByTestId('custom-modal')).not.toBeInTheDocument()
+      })
+
+      // Ensure the main publish button is disabled after successful publish
+      expect(publishButton).toBeDisabled()
+
+      // We can't easily reopen the success modal in a test environment,
+      // so we'll end the test here
+    })
+
+    test('should handle successful publish', async () => {
+      publishKmsConceptVersionMock.mockResolvedValue()
+      const { user } = setup()
+
+      // Open the publish modal
+      const publishButton = screen.getByRole('button', { name: /publish new keyword version/i })
+      await user.click(publishButton)
+
+      // Check if the modal is open by looking for the version name input
+      await waitFor(() => {
+        expect(screen.getByLabelText('Version Name:')).toBeInTheDocument()
+      })
+
+      // Enter a new version name
+      const versionNameInput = screen.getByLabelText('Version Name:')
+      await user.type(versionNameInput, 'NewVersion')
+
+      // Click the publish button in the modal
+      const modalPublishButton = screen.getByTestId('modal-action-publish')
```

Comment:
```suggestion
      const modalPublishButton = screen.getByRole('button', { name: 'Publish' })
```

Reply:
I think theres one more in this file I swapped them locally on this branch and things were still passing

Reply:
done